app/controllers/application_controller.rb

```diff
@@ -5,6 +5,145 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do 
+    if logged_in?
+      redirect to "/tweets"
+    else
+      erb :'/users/signup'
+    end
+  end
+
+  post '/signup' do 
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect to "/signup"
+    else
+      @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+      @user.save
+      session[:id] = @user.id
+      redirect to "/tweets"
+    end
+  end
+
+  get '/tweets' do 
+    if logged_in?
+      @user = current_user
+      erb :'/tweets/tweets'
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/login' do 
+    if logged_in?
+      redirect to "/tweets"
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do 
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password]) 
+      session[:id] = @user.id
+      redirect to "/tweets"
+    else
+      redirect to "/signup"
+    end
+  end
+
+  get '/logout' do
+    session.clear
+    redirect to "/login"
+  end
+
+  get '/users/:slug' do 
+    @user = User.find_by_slug(params[:slug])
+    erb :'/tweets/show_tweet'
+  end
+
+  get '/tweets/new' do 
+    if logged_in?
+      @user = current_user
+      erb :'/tweets/create_tweet'
+    else
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets' do 
+    if params[:content] != ""
+      @tweet = Tweet.create(content: params[:content])
```

Comment:
you should be able to condense line 83-85 into one line. We also don't need to set an instance variable here since we won't be accessing it in the view.

example: 

```ruby
tweet = Tweet.create(content: params[:content], user_id: current_user.id)
```

We won't need to use .save anymore as create already does this for us.